test_pull_requests

```diff
@@ -948,15 +1013,39 @@ popd
                   break
                 end
               end
+              # the main pull request and all of the co-requisite pull requests haven't been
+              # updated since the last trusted trigger, so we can begin testing
               update_evaluated_markers(repo_to_pull_request, trigger_updated_at, settings)
               submit_test_job = true
             else
+              # no trusted trigger statement has been made, so we cannot build or test this pull
+              # TODO: only pull requests with trusted triggers get process_pull_request called
```

Comment:
This logic is not getting the list of commits I think is the main difference so as to save the api calls.  There is certainly some logic that could be shared though.